specs/daito/Multi_cloud_yig_integration.md

```diff
@@ -0,0 +1,337 @@
+| sseconf | string | T | JSON (Data at rest) |
+| websiteconf | string | T | JSON |
+| tier | int32 | T | Default 1(standard) |
+
+**Object**
+
+| Column | Type | Nullable | Remark |
+| --- | --- | --- | --- |
+| key | string | F | Object key |
+| bucketid | string | F |   |
+| tenantid | string | F | Object owner id |
+| userid | string | F |   |
+| location | string | T | Backendname |
+| versionid | string | T | Random string |
+| size | int64 | T |   |
+| objectID | string | T | For backend of YIG Ceph:Random string used in ceph cluster.For other backends: the key in the backend related bucket. For example:aws\_bucket/obj1.jpg |
+| lastmodifiedtime | uint64 | T | Epoch |
+| etag | string | T | Object tags |
+| contenttype | string | T |   |
+| customattributes | string | T | JSON |
+| acl | string | T | JSON |
+| nullversion | bool | T | Used for versioning |
+| deletemarker | bool | T | Used for versioning |
+| ssetype | string | T | Data at rest (encryption) |
+| encryptionkey | string/blob | T | Data at rest (encryption) |
+| initializationvector | string/blob | T | Data at rest (encryption) |
+| tier | int32 | F | Storage tier |
+| storageClass | string | T | Storage class name |
+
+**Storage Tier**
+
+| Column | Type | Nullable | Remark |
+| --- | --- | --- | --- |
+| tier | int32 | F |   |
+| backendtype | int32 | F |   |
+| storageclass | int32 | F | Each backend has it's own definition of storage classes,and each storage class can be mapped to a specific tier in OpenSDS. |
+
+**Cluster**
+
+| Column | Type | Nullable | Remark |
+| --- | --- | --- | --- |
+| clusterid | string | F | The system to which cluster belongs to. Ex. YIG system the cluster belongs to|
+| fsid | string| F | |
+| pool | string | T | |
+| weight | int | T | |
+
+
+
+### REST API impact
+
+The API Handler won't change so REST API impact will not be there
+
+### Security impact
+
+Need to be analysed if there is any impact for the Objects in transition or in store
+
+### Other end user impact
+
+No. This is just backend change so end-user will have the same seamless experience.
+
+### Performance impact
+With new feature support and Integration of YIG and Gelato, should not impact the performance. As it will use ‘caching’, performance should improve for very large set of metadata being stored.
+The goal is to add new features without any performance impact.
+
+
+### Other deployer impact
+If OpenSDS gelato supports multiple DB, the selection and deployment of DB should be decided upon the config.
+
+### Developer impact
+
+No impact on other developers except multi-cloud.
+
+## Use Cases
+
+OpenSDS should support a scalable and enriched multi-cloud management platform. Also, it should have a performant on-premise Object Storage backend. YIG s3 compatible object storage API with a highly performant Ceph storage Backend should support multi-cloud management.
+
+### Encryption of Data at rest
+    * Regulatory compliance
+    * Data Security
+
+### Bucket Location
```

Comment:
Can you elaborate on specifying bucket location, Is it for OpenSDS virtual buckets?